diff --git a/lib/index.js b/lib/index.js
index b7ea5e30fddb9bdef0034c00f28bbd4f0dee584a..47f7717632bd55d3c42ebb39f1ef5a23c14cfa3a 100644
--- a/lib/index.js
+++ b/lib/index.js
@@ -2,8 +2,8 @@ import { upperFirst, capitalize, parseFloat as parseFloat$1, without, pick, comp
 import * as P from '@react-pdf/primitives';
 import resolveStyle, { transformColor, flatten } from '@react-pdf/stylesheet';
 import layoutEngine, { fontSubstitution, wordHyphenation, scriptItemizer, textDecoration, justification, linebreaker, bidi, fromFragments } from '@react-pdf/textkit';
-import * as Yoga from 'yoga-layout/load';
-import { loadYoga as loadYoga$1 } from 'yoga-layout/load';
+import * as Yoga from 'yoga-layout/dist/src/load.js';
+import { loadYoga as loadYoga$1 } from 'yoga-layout/dist/src/load.js';
 import emojiRegex from 'emoji-regex-xs';
 import resolveImage from '@react-pdf/image';
 
@@ -2998,7 +2998,7 @@ const splitPage = (page, pageNumber, fontStore, yoga) => {
     const dynamicPage = resolveDynamicPage({ pageNumber }, page, fontStore, yoga);
     const height = page.style.height;
     const [currentChilds, nextChilds] = splitNodes(wrapArea, contentArea, dynamicPage.children);
-    const relayout = (node) => 
+    const relayout = (node) =>
     // @ts-expect-error rework pagination
     relayoutPage(node, fontStore, yoga);
     const currentBox = { ...page.box, height };
