diff --git a/mongodb.d.ts b/mongodb.d.ts
index 9696a0d0104095e8e4dfa4a4fe05fe81fd8b50c7..f5aa6e76fe6f2e6a55e6e2a720be15b7a6f98107 100644
--- a/mongodb.d.ts
+++ b/mongodb.d.ts
@@ -5535,7 +5535,7 @@ export declare interface MonitorOptions extends Omit<ConnectionOptions, 'id' | '
  * depth any helpers that make use of NestedPaths should devolve to not asserting any
  * type safety on the input.
  */
-export declare type NestedPaths<Type, Depth extends number[]> = Depth['length'] extends 8 ? [] : Type extends string | number | boolean | Date | RegExp | Buffer | Uint8Array | ((...args: any[]) => any) | {
+export declare type NestedPaths<Type, Depth extends number[]> = Depth['length'] extends 3 ? [] : Type extends string | number | boolean | Date | RegExp | Buffer | Uint8Array | ((...args: any[]) => any) | {
     _bsontype: string;
 } ? [] : Type extends ReadonlyArray<infer ArrayType> ? [] | [number, ...NestedPaths<ArrayType, [...Depth, 1]>] : Type extends Map<string, any> ? [string] : Type extends object ? {
     [Key in Extract<keyof Type, string>]: Type[Key] extends Type ? [Key] : Type extends Type[Key] ? [Key] : Type[Key] extends ReadonlyArray<infer ArrayType> ? Type extends ArrayType ? [Key] : ArrayType extends Type ? [Key] : [
