// presence_service/baseService.js
//
// Autogenerated by Thrift Compiler (0.9.1)
//
// DO NOT EDIT UNLESS YOU ARE SURE THAT YOU KNOW WHAT YOU ARE DOING
//
var Thrift = require('thrift').Thrift;

var ttypes = require('./sharedService_types');
//HELPER FUNCTIONS AND STRUCTURES

baseService_ping_args = function(args) {
  this.answer = 42;
  if (args) {
    if (args.answer !== undefined) {
      this.answer = args.answer;
    }
  }
};
baseService_ping_args.prototype = {};
baseService_ping_args.prototype.read = function(input) {
  input.readStructBegin();
  while (true)
  {
    var ret = input.readFieldBegin();
    var fname = ret.fname;
    var ftype = ret.ftype;
    var fid = ret.fid;
    if (ftype == Thrift.Type.STOP) {
      break;
    }
    switch (fid)
    {
      case 1:
      if (ftype == Thrift.Type.I32) {
        this.answer = input.readI32();
      } else {
        input.skip(ftype);
      }
      break;
      case 0:
        input.skip(ftype);
        break;
      default:
        input.skip(ftype);
    }
    input.readFieldEnd();
  }
  input.readStructEnd();
  return;
};

baseService_ping_args.prototype.write = function(output) {
  output.writeStructBegin('baseService_ping_args');
  if (this.answer !== null && this.answer !== undefined) {
    output.writeFieldBegin('answer', Thrift.Type.I32, 1);
    output.writeI32(this.answer);
    output.writeFieldEnd();
  }
  output.writeFieldStop();
  output.writeStructEnd();
  return;
};

baseService_ping_result = function(args) {
  this.success = null;
  this.serr = null;
  if (args instanceof ttypes.ServerError) {
    this.serr = args;
    return;
  }
  if (args) {
    if (args.success !== undefined) {
      this.success = args.success;
    }
    if (args.serr !== undefined) {
      this.serr = args.serr;
    }
  }
};
baseService_ping_result.prototype = {};
baseService_ping_result.prototype.read = function(input) {
  input.readStructBegin();
  while (true)
  {
    var ret = input.readFieldBegin();
    var fname = ret.fname;
    var ftype = ret.ftype;
    var fid = ret.fid;
    if (ftype == Thrift.Type.STOP) {
      break;
    }
    switch (fid)
    {
      case 0:
      if (ftype == Thrift.Type.I32) {
        this.success = input.readI32();
      } else {
        input.skip(ftype);
      }
      break;
      case 1:
      if (ftype == Thrift.Type.STRUCT) {
        this.serr = new ttypes.ServerError();
        this.serr.read(input);
      } else {
        input.skip(ftype);
      }
      break;
      default:
        input.skip(ftype);
    }
    input.readFieldEnd();
  }
  input.readStructEnd();
  return;
};

baseService_ping_result.prototype.write = function(output) {
  output.writeStructBegin('baseService_ping_result');
  if (this.success !== null && this.success !== undefined) {
    output.writeFieldBegin('success', Thrift.Type.I32, 0);
    output.writeI32(this.success);
    output.writeFieldEnd();
  }
  if (this.serr !== null && this.serr !== undefined) {
    output.writeFieldBegin('serr', Thrift.Type.STRUCT, 1);
    this.serr.write(output);
    output.writeFieldEnd();
  }
  output.writeFieldStop();
  output.writeStructEnd();
  return;
};

baseService_validate_param_args = function(args) {
  this.token = null;
  this.field = null;
  this.value = null;
  this.params = null;
  if (args) {
    if (args.token !== undefined) {
      this.token = args.token;
    }
    if (args.field !== undefined) {
      this.field = args.field;
    }
    if (args.value !== undefined) {
      this.value = args.value;
    }
    if (args.params !== undefined) {
      this.params = args.params;
    }
  }
};
baseService_validate_param_args.prototype = {};
baseService_validate_param_args.prototype.read = function(input) {
  input.readStructBegin();
  while (true)
  {
    var ret = input.readFieldBegin();
    var fname = ret.fname;
    var ftype = ret.ftype;
    var fid = ret.fid;
    if (ftype == Thrift.Type.STOP) {
      break;
    }
    switch (fid)
    {
      case 1:
      if (ftype == Thrift.Type.STRING) {
        this.token = input.readString();
      } else {
        input.skip(ftype);
      }
      break;
      case 2:
      if (ftype == Thrift.Type.STRING) {
        this.field = input.readString();
      } else {
        input.skip(ftype);
      }
      break;
      case 3:
      if (ftype == Thrift.Type.STRING) {
        this.value = input.readString();
      } else {
        input.skip(ftype);
      }
      break;
      case 4:
      if (ftype == Thrift.Type.STRUCT) {
        this.params = new ttypes.ORMFilter();
        this.params.read(input);
      } else {
        input.skip(ftype);
      }
      break;
      default:
        input.skip(ftype);
    }
    input.readFieldEnd();
  }
  input.readStructEnd();
  return;
};

baseService_validate_param_args.prototype.write = function(output) {
  output.writeStructBegin('baseService_validate_param_args');
  if (this.token !== null && this.token !== undefined) {
    output.writeFieldBegin('token', Thrift.Type.STRING, 1);
    output.writeString(this.token);
    output.writeFieldEnd();
  }
  if (this.field !== null && this.field !== undefined) {
    output.writeFieldBegin('field', Thrift.Type.STRING, 2);
    output.writeString(this.field);
    output.writeFieldEnd();
  }
  if (this.value !== null && this.value !== undefined) {
    output.writeFieldBegin('value', Thrift.Type.STRING, 3);
    output.writeString(this.value);
    output.writeFieldEnd();
  }
  if (this.params !== null && this.params !== undefined) {
    output.writeFieldBegin('params', Thrift.Type.STRUCT, 4);
    this.params.write(output);
    output.writeFieldEnd();
  }
  output.writeFieldStop();
  output.writeStructEnd();
  return;
};

baseService_validate_param_result = function(args) {
  this.success = null;
  this.serr = null;
  if (args instanceof ttypes.ServerError) {
    this.serr = args;
    return;
  }
  if (args) {
    if (args.success !== undefined) {
      this.success = args.success;
    }
    if (args.serr !== undefined) {
      this.serr = args.serr;
    }
  }
};
baseService_validate_param_result.prototype = {};
baseService_validate_param_result.prototype.read = function(input) {
  input.readStructBegin();
  while (true)
  {
    var ret = input.readFieldBegin();
    var fname = ret.fname;
    var ftype = ret.ftype;
    var fid = ret.fid;
    if (ftype == Thrift.Type.STOP) {
      break;
    }
    switch (fid)
    {
      case 0:
      if (ftype == Thrift.Type.BOOL) {
        this.success = input.readBool();
      } else {
        input.skip(ftype);
      }
      break;
      case 1:
      if (ftype == Thrift.Type.STRUCT) {
        this.serr = new ttypes.ServerError();
        this.serr.read(input);
      } else {
        input.skip(ftype);
      }
      break;
      default:
        input.skip(ftype);
    }
    input.readFieldEnd();
  }
  input.readStructEnd();
  return;
};

baseService_validate_param_result.prototype.write = function(output) {
  output.writeStructBegin('baseService_validate_param_result');
  if (this.success !== null && this.success !== undefined) {
    output.writeFieldBegin('success', Thrift.Type.BOOL, 0);
    output.writeBool(this.success);
    output.writeFieldEnd();
  }
  if (this.serr !== null && this.serr !== undefined) {
    output.writeFieldBegin('serr', Thrift.Type.STRUCT, 1);
    this.serr.write(output);
    output.writeFieldEnd();
  }
  output.writeFieldStop();
  output.writeStructEnd();
  return;
};

baseServiceClient = exports.Client = function(output, pClass) {
    this.output = output;
    this.pClass = pClass;
    this.seqid = 0;
    this._reqs = {};
};
baseServiceClient.prototype = {};
baseServiceClient.prototype.ping = function(answer, callback) {
  this.seqid += 1;
  this._reqs[this.seqid] = callback;
  this.send_ping(answer);
};

baseServiceClient.prototype.send_ping = function(answer) {
  var output = new this.pClass(this.output);
  output.writeMessageBegin('ping', Thrift.MessageType.CALL, this.seqid);
  var args = new baseService_ping_args();
  args.answer = answer;
  args.write(output);
  output.writeMessageEnd();
  return this.output.flush();
};

baseServiceClient.prototype.recv_ping = function(input,mtype,rseqid) {
  var callback = this._reqs[rseqid] || function() {};
  delete this._reqs[rseqid];
  if (mtype == Thrift.MessageType.EXCEPTION) {
    var x = new Thrift.TApplicationException();
    x.read(input);
    input.readMessageEnd();
    return callback(x);
  }
  var result = new baseService_ping_result();
  result.read(input);
  input.readMessageEnd();

  if (null !== result.serr) {
    return callback(result.serr);
  }
  if (null !== result.success) {
    return callback(null, result.success);
  }
  return callback('ping failed: unknown result');
};
baseServiceClient.prototype.validate_param = function(token, field, value, params, callback) {
  this.seqid += 1;
  this._reqs[this.seqid] = callback;
  this.send_validate_param(token, field, value, params);
};

baseServiceClient.prototype.send_validate_param = function(token, field, value, params) {
  var output = new this.pClass(this.output);
  output.writeMessageBegin('validate_param', Thrift.MessageType.CALL, this.seqid);
  var args = new baseService_validate_param_args();
  args.token = token;
  args.field = field;
  args.value = value;
  args.params = params;
  args.write(output);
  output.writeMessageEnd();
  return this.output.flush();
};

baseServiceClient.prototype.recv_validate_param = function(input,mtype,rseqid) {
  var callback = this._reqs[rseqid] || function() {};
  delete this._reqs[rseqid];
  if (mtype == Thrift.MessageType.EXCEPTION) {
    var x = new Thrift.TApplicationException();
    x.read(input);
    input.readMessageEnd();
    return callback(x);
  }
  var result = new baseService_validate_param_result();
  result.read(input);
  input.readMessageEnd();

  if (null !== result.serr) {
    return callback(result.serr);
  }
  if (null !== result.success) {
    return callback(null, result.success);
  }
  return callback('validate_param failed: unknown result');
};
baseServiceProcessor = exports.Processor = function(handler) {
  this._handler = handler
}
baseServiceProcessor.prototype.process = function(input, output) {
  var r = input.readMessageBegin();
  if (this['process_' + r.fname]) {
    return this['process_' + r.fname].call(this, r.rseqid, input, output);
  } else {
    input.skip(Thrift.Type.STRUCT);
    input.readMessageEnd();
    var x = new Thrift.TApplicationException(Thrift.TApplicationExceptionType.UNKNOWN_METHOD, 'Unknown function ' + r.fname);
    output.writeMessageBegin(r.fname, Thrift.MessageType.Exception, r.rseqid);
    x.write(output);
    output.writeMessageEnd();
    output.flush();
  }
}

baseServiceProcessor.prototype.process_ping = function(seqid, input, output) {
  var args = new baseService_ping_args();
  args.read(input);
  input.readMessageEnd();
  this._handler.ping(args.answer, function (err, result) {
    var result = new baseService_ping_result((err != null ? err : {success: result}));
    output.writeMessageBegin("ping", Thrift.MessageType.REPLY, seqid);
    result.write(output);
    output.writeMessageEnd();
    output.flush();
  })
}

baseServiceProcessor.prototype.process_validate_param = function(seqid, input, output) {
  var args = new baseService_validate_param_args();
  args.read(input);
  input.readMessageEnd();
  this._handler.validate_param(args.token, args.field, args.value, args.params, function (err, result) {
    var result = new baseService_validate_param_result((err != null ? err : {success: result}));
    output.writeMessageBegin("validate_param", Thrift.MessageType.REPLY, seqid);
    result.write(output);
    output.writeMessageEnd();
    output.flush();
  })
}
